plugins/services/src/js/reducers/serviceForm/FormReducers/Volumes.ts

```diff
@@ -20,14 +39,17 @@ export function FormReducer(state = [], { type, path, value }) {
         state.push({ ...(value || defaultVolume) });
         break;
       case REMOVE_ITEM:
-        state = state.filter((item, index) => index !== value);
+        state = state.filter((_, index) => index !== value);
         break;
     }
     return state;
   }
 
   if (joinedPath.includes("volumes") && type === SET) {
     const index = joinedPath.match(/\d+/)[0];
+    if (`volumes.${index}` === joinedPath) {
+      state[index] = value;
+    }
```

Comment:
this is the main trick to get rid of having to implement a reducer-path for every single property. we just set the whole volume at once all the time and carefully (partly via type inference) handle object manipulation.